packages/abstractions/src/recordWithCaseInsensitiveKeys.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * A Proxy that allows implementation of a Object with case insensitive keys
+ * @param canon 
+ * @returns 
+ */
+function dictionaryWithCanonicalKeys<V>(canon: (prop: keyof any) => keyof any) {
+  return new Proxy<{ [k: string]: V }>(
+    {},
+    {
+      get(target, prop) {
```

Comment:
I have added some comments on the code.

Reply:
right but that still doesn't answer my question.
If I send a "param1" and a "Param1" what will happen? are the reflect mnethods normalizing the target parameter?

Reply:
I probably didn't understand your question well, my understanding is the conversion of the key will happen at:
``` javascript
const record: Record<string, T> = dictionaryWithCanonicalKeys<T>(
    p => (typeof p === "string" ? p.toLowerCase() : p)
);
  ```
If the code looks confusing, the other thing I can do is maybe move the Proxy into initialization method. It would look something like this:
  ``` Javascript
  function createCaseInsensitiveProxy<T>(): CaseInsensitiveRecord<T> {
  return new Proxy({}, {
    get(target, name) {
      const key = Object.keys(target).find(k => k.toLowerCase() === name.toString().toLowerCase());
      return target[key!];
    },
    set(target, name, value) {
      const key = Object.keys(target).find(k => k.toLowerCase() === name.toString().toLowerCase());
      target[key!] = value;
      return true;
    }
    //....
  });
}
  ```

Reply:
Is there a test case that I can use to capture this scenario? Also, would the alternate proxy snippet above be any better?

Reply:
it's better in terms of functionality but probably much slower o(n) vs what I imagine to be o(1) previously.

Any template with a non all lower case parameter name would do, e.g. `https:://foo.bar{?MyParam}`